python/ray/actor.py

```diff
@@ -543,6 +543,12 @@ def __init__(
             modified_class, actor_creation_function_descriptor
         )
 
+    def __getstate__(self):
+        # `last_export_session_and_job` is worker-local. Reset it when pickling.
+        state = dict(self.__dict__)
+        state["last_export_session_and_job"] = None
```

Comment:
Even if we pickle it, the next job will have a different job id so 

```
 if not meta.is_cross_language and (
            meta.last_export_session_and_job != worker.current_session_and_job
        ):
            # If this actor class was not exported in this session and job,
            # we need to export this function again, because current GCS
            # doesn't have it.

            # After serialize / deserialize modified class, the __module__
            # of modified class will be ray.cloudpickle.cloudpickle.
            # So, here pass actor_creation_function_descriptor to make
            # sure export actor class correct.
            worker.function_actor_manager.export_actor_class(
                meta.modified_class,
                meta.actor_creation_function_descriptor,
                meta.method_meta.methods.keys(),
            )
            meta.last_export_session_and_job = worker.current_session_and_job
```

this if statement will be True and we will still export it to GCS?

Reply:
I discovered this in a unit test. In first run, it's Job 01000000, and when we call `ray.shutdown()` the whole ray cluster exited, and in the next run it's still 01000000. So both times, session_and_job is `(0, 01000000)`.